fullstop-plugins/fullstop-keypair-plugin/src/main/java/org/zalando/stups/fullstop/plugin/keypair/KeyPairPlugin.java

```diff
@@ -69,7 +71,8 @@ public void processEvent(final CloudTrailEvent event) {
 
         List<String> keyNames = CloudtrailEventSupport.containsKeyNames(event.getEventData().getRequestParameters());
         if (!CollectionUtils.isEmpty(keyNames)) {
-            violationStore.save(String.format("KeyPair must be blank, but was %s", keyNames));
+            violationStore.save(new ViolationStoreDomain(event.getEventData().getAccountId(),
+                    event.getEventData().getAwsRegion(), format("KeyPair must be blank, but was %s", keyNames)));
```

Comment:
Maybe you should use CloudTrailEventSupport and its static methods to just write 'getRegion(event)'.